style/git/README.md

```diff
@@ -179,9 +195,60 @@ scope: {
   myAccessor: '=' // in directive's template change myAccessor() to myAccessor
 }
 
-The removed `inject` wasn't generaly useful for directives so there should be no code using it.
+The removed `inject` wasn't generally useful for directives so there should be no code using it.
 ```
 
+The Sparkbox Git Flow
+---------------------
+
+Every place you work will have a different Git flow. At Sparkbox the Git flow is as follows:
+
+1. Create a branch off of master to fix an issue assigned to you.
+
+1. Push this branch up to GitHub, and create a pull request when you are ready.
+
+1. Assign a reviewer to your pull request. Leave some nice documentation or even a [screencast](https://viewedit.com/) to give your reviewer a bit more background about the code you wrote in this pull request (include information about any interesting or difficult code, and where the reviewer can see the code that you are referencing).
+
+1. Your reviewer will either approve or request changes on your pull request.  
+    - If they request changes, please implement or further discuss their comments with them.
+
+1. If your reviewer approves your pull request, they will *rebase* your branch on top of master (see my description of rebasing in the next section for more information).
+
+Rebasing a Branch onto the Master Branch
+----------------------------------------------
+
+Rather than merging commits from another branch into the master branch, Sparkbox prefers to rebase commits on top of the master branch.  Here is a diagram that depicts the differences between merging and rebasing:
+
+![Merging](merge.png)
+
+![Rebasing](rebase.png)
+
+For the purposes of this example explaining rebasing, let's say that the branch you need to rebase is called `chore-teaching-example`, and that this branch has gone through all of the Git flow discussed above for a pull request.  Follow these next steps to rebase a branch onto master:
+
+1. Open your terminal and checkout the branch to be rebased: `git checkout chore-teaching-example`.
```

Comment:
You should probably do a `git fetch` &/or `git pull` first. If you don't, you might be rebasing again after all is said and done, because your `origin/master` and `master` are out of date. This can happen anyways if someone pushes while you're rebasing, but ¯\_(ツ)_/¯